ql/src/java/org/apache/hadoop/hive/ql/exec/repl/ReplLoadTask.java

```diff
@@ -292,19 +306,30 @@ private void setUpDependencies(TaskTracker parentTasks, TaskTracker childTasks)
     }
   }
 
-  private void createBuilderTask(List<Task<? extends Serializable>> rootTasks,
-      boolean shouldCreateAnotherLoadTask) {
-  /*
-    use loadTask as dependencyCollection
-   */
-    if (shouldCreateAnotherLoadTask) {
-      Task<ReplLoadWork> loadTask = TaskFactory.get(work, conf);
-      DAGTraversal.traverse(rootTasks, new AddDependencyToLeaves(loadTask));
-    }
+  private void createBuilderTask(List<Task<? extends Serializable>> rootTasks) {
+    // Use loadTask as dependencyCollection
+    Task<ReplLoadWork> loadTask = TaskFactory.get(work, conf);
+    DAGTraversal.traverse(rootTasks, new AddDependencyToLeaves(loadTask));
   }
 
   @Override
   public StageType getType() {
-    return StageType.REPL_BOOTSTRAP_LOAD;
+    return work.isIncrementalLoad() ? StageType.REPL_INCREMENTAL_LOAD : StageType.REPL_BOOTSTRAP_LOAD;
+  }
+
+  private int executeIncrementalLoad(DriverContext driverContext) {
+    try {
+      IncrementalLoad load = work.getIncrementalLoad();
+      childTasks = Collections.singletonList(load.execute(driverContext, getHive(), LOG));
```

Comment:
Use this.childTasks to be precise.

Reply:
for what precision ?

Reply:
Just for readability that it is class variable. Not mandatory though.